hack/kube-api-linter.yaml

```diff
@@ -0,0 +1,50 @@
+run:
+  timeout: 10m
+  go: "1.23"
+  allow-parallel-runners: true
+
+linters:
+  disable-all: true
+  enable:
+    - kubeapilinter # linter for Kube API conventions
+   
+linters-settings:
+  custom:
+    kubeapilinter:
+      type: "module"
+      description: kube-api-linter and lints Kube like APIs based on API conventions and best practices.
+      settings:
+        linters:
+          enable:
+          # - "commentstart" # Ensure comments start with the serialized version of the field name.
+          # - "conditions" # Ensure conditions have the correct json tags and markers.
+          # - "integers" # Ensure only int32 and int64 are used for integers.
+          # - "jsontags" # Ensure every field has a json tag.
+          # - "maxlength" # Ensure all strings and arrays have maximum lengths/maximum items. ONLY for CRDs until declarative markers exist in core types.
+          # - "nobools" # Bools do not evolve over time, should use enums instead.
+          # - "nofloats" # Ensure floats are not used.
+          # - "nomaps" # Ensure maps are not used, unless they are `map[string]string` (for labels/annotations/etc).
+          # - "nophase" # Ensure field names do not have the word "phase" in them.
```

Comment:
I've left these commented so we know what's implemented and still to be enabled in future PRs.

That said, there are many rules still to enable.